analysis/deathknight/src/shadowlands/conduits/EternalHunger.tsx

```diff
@@ -0,0 +1,73 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import calculateEffectiveDamage from 'parser/core/calculateEffectiveDamage';
+import Events, { DamageEvent, SummonEvent } from 'parser/core/Events';
+import ConduitSpellText from 'parser/ui/ConduitSpellText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+const ETERNAL_HUNGER_BASE_INCREASE = 0.054;
+const ETERNAL_HUNGER_PER_RANK = 0.006;
+
+class EternalHunger extends Analyzer {
+  conduitRank: number = 0;
+  addedDamage: number = 0;
+  increase!: number;
+  pets: Array<{ sourceID: number; sourceInstance: number }> = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.conduitRank = this.selectedCombatant.conduitRankBySpellID(SPELLS.ETERNAL_HUNGER.id);
+
+    if (!this.conduitRank) {
+      this.active = false;
+      return;
+    }
+    this.increase = ETERNAL_HUNGER_BASE_INCREASE + this.conduitRank * ETERNAL_HUNGER_PER_RANK;
+
+    // Can't filter this .by(SELECTED_PLAYER_PET) since that only includes pets present
```

Comment:
I know this module was made for blood, but I think you need a listener for .by(SELECTED_PLAYER_PET) in conjunction with your summons tracking here.  The permanent unholy pet does not have a summon event in normal gameplay and so its damage will never be picked up.

Reply:
There's 2 conditions in onDamage: `this.isPet(event) || this.owner.byPlayerPet(event)` the second one should account for permanent pets. I looked at a couple unholy logs and they're calculating about 3x as much dps from the conduit as my Blood logs are, so gut check passes. Did you have a log you were worried about?

Reply:
I did some more looking into the numbers and EH is just not as good at higher ranks as I thought it was.  It was coming out at 5-6% on unholy logs.  Comparing logs with 5 minute kill times to sims shows a similar percentage, close enough that the permanent pet would have to be picked up by wowa.